oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/JwtKafkaPrincipal.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.server.authorizer;
+
+import io.strimzi.kafka.oauth.common.BearerTokenWithPayload;
+import org.apache.kafka.common.security.auth.KafkaPrincipal;
+
+public class JwtKafkaPrincipal extends KafkaPrincipal {
+
+    private final BearerTokenWithPayload jwt;
+
+    public JwtKafkaPrincipal(String principalType, String name) {
+        this(principalType, name, null);
+    }
+
+    public JwtKafkaPrincipal(String principalType, String name, BearerTokenWithPayload jwt) {
+        super(principalType, name);
+        this.jwt = jwt;
+    }
+
+    public BearerTokenWithPayload getJwt() {
+        return jwt;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        // We don't care about jwt field for equals comparison
+        return super.equals(o);
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode();
+    }
```

Comment:
Why are we overriding these if we're just delegating to super?

Reply:
Maybe just my lazy way of dealing with SpotBugs. I would actually prefer to add SpotBugs exception. Not yet quite versed in how to decide each time.
I'll gladly add SpotBugs exception.

Reply:
Maybe I'm being stupid: Does the superclass do an `instanceof` or does it check class equality?

Reply:
In fact it's not a good practice to do any assumptions about super equals() that is not part of our code base. I'll do a proper equals impl.